### PR TITLE
track signed-in state via Analytics

### LINKED ISSUE #1941: Not properly tracking signed in state.
I noticed when we moved off of DevSite that we stopped tracking when the user is signed in/out.

We have the ability to track signed in state by setting an analytics custom dimension:
```
ga.set(site.analytics.dimensions.SIGNED_IN, true);
```
This means all subsequent pings will contain this info so we can discern actions that occurred from signed in versus signed out users.

I'd suggest that we move the code we currently use to set the signed in state out of fb.js, and into actions.js. Then we can set the custom dimension as a side effect.

Phil suggested that we also consider delaying the send event for our initial pageview, and wait until firebase auth has returned so we can track whether the user was signed in/out for the initial page view. He suggested we set a timeout to ensure we capture the pageview event if fb is taking too long to return.

### PR DESCRIPTION
**Pull request by samthor**
Fixes #1941.

Adds an additional "basic" script which is run by everyone. This lets us share the dimensions rather than hard-coding them inline.

### CODE REVIEW

**Comment by samthor** [comment]
I don't know. I'll revert it.

File: src/site/_includes/layout.njk [hunk 1]
```diff
@@ -44,14 +44,12 @@
...
-      ga('set', 'transport', 'beacon');
-      ga('set', '{{ site.analytics.dimensions.TRACKING_VERSION }}', '{{ site.analytics.version }}');
-      ga('send', 'pageview');
-    </script>
     <script async src="//www.google-analytics.com/analytics.js"></script>
+    {#
+      The basic script is "defer" as it should run before the normal bootstrap module code (which
+      is defer by default).
+    #}
+    <script defer src="/basic.js"></script>
```
Comment (robdodson) [inline src/site/_includes/layout.njk:52]: wdyt about renaming the file to something more descriptive like `analytics`? although i think we already have an analytics.js elsewhere, but `basic` seems like folks wouldn't be able to tell at a glance what it does.

File: build.js [hunk 2]
```diff
@@ -42,6 +42,11 @@ process.on('unhandledRejection', (reason, p) => {
  * Virtual imports made available to all bundles. Used for site config and globals.
  */
 const virtualImports = {
+  webdev_analytics: {
```
Comment (robdodson) [inline build.js:45]: why put these here instead of in `webdev_config`?

File: src/lib/basic.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+/**
+ * @fileoverview Run by all browsers in `script defer`. Used to set up
+ * Analytics, including with our known previous signed-in state that was cached
+ * in `localStorage`.
+ *
+ * This file is built with Rollup, but separately to our core bundle.
```
Comment (robdodson) [inline src/lib/basic.js:6]: Is it worth mentioning explicitly that this file should never import anything from our core bundle?

Also might be good to explicitly state it's for IE11 etc.

File: build.js (hunk 2 above)
Comment (samthor) [reply build.js:45]: You don't need to import everything if you only want a subset of config. Importing `{* as foo}` can bring all values in, so it's nice to break it apart into different configs.

**Pull request merged**